File: burp/src/main/java/burp/BHttpRequestResponse.java
package burp;

import org.apache.commons.codec.binary.Base64;


public class BHttpRequestResponse extends BSocketMessage implements IHttpRequestResponse {
    public BHttpRequest request;
    public BHttpResponse response;

    public byte[] getRequest() {
        return Base64.decodeBase64(request.raw);
    }

    public void setRequest(byte[] message) {
        request.raw = Base64.encodeBase64String(message);
    }

    public byte[] getResponse() {
        return Base64.decodeBase64(response.raw);
    }

    public void setResponse(byte[] message) {
        response.raw = Base64.encodeBase64String(message);
    }

    public String getComment() {
        return request.comment;
    }

    public void setComment(String comment) {
        request.comment = comment;
        response.comment = comment;
    }

    public String getHighlight() {
        return request.highlight;
    }

    public void setHighlight(String color) {
        request.highlight = color;
        response.highlight = color;
    }

    public IHttpService getHttpService() {
        return new IHttpService() {
            @Override
            public String getHost() {
                return request.host;
            }

            @Override
            public int getPort() {
                return request.port;
            }

            @Override
            public String getProtocol() {
                return  request.protocol;
            }
        };
    }

    public void setHttpService(IHttpService httpService) {
        request.host = httpService.getHost();
        request.port = httpService.getPort();
        request.protocol = httpService.getProtocol();
        response.host = httpService.getHost();
        response.port = httpService.getPort();
        response.protocol = httpService.getProtocol();
    }
}

File: src/main/kotlin/burp/BHttpRequestResponse.kt
package burp

import java.util.Base64

class BHttpRequestResponse(val httpMessage: HttpRequestResponse, val service: HttpService): IHttpRequestResponse {
    override fun setComment(comment: String) {
        httpMessage.comment = comment
    }

    override fun getComment(): String {
        return httpMessage.comment
    }

    override fun setRequest(rawRequest: ByteArray) {
        // TODO: I'm not sure if i should set all the other variables for a request.
        httpMessage.request.raw = Base64.getEncoder().encodeToString(rawRequest)
    }

    override fun getHttpService(): IHttpService {
        return BHttpService(service)
    }

    override fun getHighlight(): String {
        return highlight
    }

    override fun getResponse(): ByteArray {
        return Base64.getDecoder().decode(httpMessage.response.raw)
    }

    override fun setHighlight(color: String) {
        httpMessage.highlight = color
    }

    override fun setHttpService(httpService: IHttpService) {
        service.protocol = httpService.protocol
        service.port = httpService.port
        service.host = httpService.host
    }

    override fun setResponse(rawResponse: ByteArray) {
        httpMessage.response.raw = Base64.getEncoder().encodeToString(rawResponse)
    }

    override fun getRequest(): ByteArray {
        return Base64.getDecoder().decode(httpMessage.request.raw)
    }
}